portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,37 +43,64 @@
 /** Servlet that store and return information about user comments. */
 @WebServlet("/comments")
 public class DataServlet extends HttpServlet {
-
-  // A list of all comments.
-  ArrayList<Comment> commentsList = new ArrayList<>();
-
   /** Processes GET requests for "/comments" and returns a list of comments in JSON format. */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = convertToJsonList();
+    // Get requested number of comments.
+    int numberOfComments = Integer.valueOf(request.getParameter("number"));
```

Comment:
This will throw an exception if the parameter is not specified (or an empty or non-numeric value is specified), which it turn would return a 500 Internal Server error to the client without telling them how to fix it. It is usually better to either fall back to a default value (e.g catch the exception and set the value to -1), or specify more detail as part of the error so the client would have the ability to fix the error.